File: Java/DesignPatterns/src/main/java/designPatterns/behavioral/observer/MainObserver.java
package designPatterns.behavioral.observer;

public class MainObserver {
  public static void main(String[] args) {
    Topic topic = new Topic();
    Observer obj1 = new TopicSubscriber("Subscriber1");
    Observer obj2 = new TopicSubscriber("Subscriber2");
    Observer obj3 = new TopicSubscriber("Subscriber3");
    topic.register(obj1);
    topic.register(obj2);
    topic.register(obj3);

    System.out.println(topic.postMessage("Hello Subscribers!"));
  }
}

File: Kotlin/DesignPatterns/src/main/kotlin/behavioral/observer/MainObserver.kt
package behavioral.observer

fun main() {
  var topic = Topic()
  var obj1 = TopicSubscriber("Subscriber1")
  var obj2 = TopicSubscriber("Subscriber2")
  var obj3 = TopicSubscriber("Subscriber3")
  topic.register(obj1)
  topic.register(obj2)
  topic.register(obj3)

  println(topic.postMessage("Hello Subscribers!"))
}